examples/storm-starter/src/jvm/storm/starter/SlidingWindowTopology.java

```diff
@@ -0,0 +1,129 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package storm.starter;
+
+import backtype.storm.Config;
+import backtype.storm.LocalCluster;
+import backtype.storm.StormSubmitter;
+import backtype.storm.spout.SpoutOutputCollector;
+import backtype.storm.task.OutputCollector;
+import backtype.storm.task.TopologyContext;
+import backtype.storm.topology.OutputFieldsDeclarer;
+import backtype.storm.topology.TopologyBuilder;
+import backtype.storm.topology.base.BaseRichSpout;
+import backtype.storm.topology.base.BaseWindowedBolt;
+import backtype.storm.tuple.Fields;
+import backtype.storm.tuple.Tuple;
+import backtype.storm.tuple.Values;
+import backtype.storm.utils.Utils;
+import backtype.storm.windowing.TupleWindow;
+import storm.starter.bolt.PrinterBolt;
+
+import java.util.Map;
+import java.util.Random;
+
+import static backtype.storm.topology.base.BaseWindowedBolt.Count;
+import static backtype.storm.topology.base.BaseWindowedBolt.Duration;
+
+/**
+ * A sample topology that demonstrates the usage of {@link backtype.storm.topology.IWindowedBolt}
+ * to calculate sliding window sum.
+ */
+public class SlidingWindowTopology {
+
+    /*
+     * emits random integers every 100 ms
+     */
+    private static class RandomIntegerSpout extends BaseRichSpout {
+        SpoutOutputCollector collector;
+
+        @Override
+        public void declareOutputFields(OutputFieldsDeclarer declarer) {
+            declarer.declare(new Fields("value"));
+        }
+
+        @Override
+        public void open(Map conf, TopologyContext context, SpoutOutputCollector collector) {
+            this.collector = collector;
+        }
+
+        @Override
+        public void nextTuple() {
+            Utils.sleep(100);
+            Random rand = new Random();
+            Integer value = rand.nextInt(1000);
+            collector.emit(new Values(value));
+        }
+    }
+
+    /*
+     * Computes sliding window sum
+     */
+    private static class SlidingWindowSumBolt extends BaseWindowedBolt {
+        private int sum = 0;
+        private OutputCollector collector;
+
+        @Override
+        public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+            this.collector = collector;
+        }
+
+        @Override
+        public void execute(TupleWindow inputWindow) {
+            System.out.println("Events in current window: " + inputWindow.get().size());
+            for (Tuple tuple : inputWindow.getNew()) {
+                sum += (int) tuple.getValue(0);
+            }
+            for (Tuple tuple : inputWindow.getExpired()) {
+                sum -= (int) tuple.getValue(0);
+            }
+            collector.emit(new Values(sum));
```

Comment:
Oh I see you have the `getNew()` and the `getExpired()`.  OK It would be good to add a comment on that, because it was very confusing to me when I just glanced through the code.